src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-chart/cephfs-chart.component.ts

```diff
@@ -22,136 +23,139 @@ export class CephfsChartComponent implements OnChanges, OnInit {
     );
-    chartTooltip.getTitle = getTitle;
+    chartTooltip.getTitle = (ts) => moment(ts, 'x').format('LTS');
     chartTooltip.checkOffset = true;
-
-    const lhsData = this.convert_timeseries(this.mdsCounter[this.lhsCounter]);
-    const rhsData = this.delta_timeseries(this.mdsCounter[this.rhsCounter]);
-
-    this.chart = {
-      datasets: [
-        {
-          label: this.lhsCounter,
-          yAxisID: 'LHS',
-          data: lhsData,
-          tension: 0.1
-        },
-        {
-          label: this.rhsCounter,
-          yAxisID: 'RHS',
-          data: rhsData,
-          tension: 0.1
-        }
-      ],
+    _.merge(this.chart, {
       options: {
         title: {
-          text: this.mdsCounter.name,
-          display: true
-        },
-        responsive: true,
-        maintainAspectRatio: false,
-        legend: {
-          position: 'top'
-        },
-        scales: {
-          xAxes: [
-            {
-              position: 'top',
-              type: 'time',
-              time: {
-                displayFormats: {
-                  quarter: 'MMM YYYY'
-                }
-              },
-              ticks: {
-                maxRotation: 0
-              }
-            }
-          ],
-          yAxes: [
-            {
-              id: 'LHS',
-              type: 'linear',
-              position: 'left',
-              min: 0
-            },
-            {
-              id: 'RHS',
-              type: 'linear',
-              position: 'right',
-              min: 0
-            }
-          ]
+          text: this.mdsCounter.name
         },
         tooltips: {
-          enabled: false,
-          mode: 'index',
-          intersect: false,
-          position: 'nearest',
-          callbacks: {
-            // Pick the Unix timestamp of the first tooltip item.
-            title: function(tooltipItems, data) {
-              let ts = 0;
-              if (tooltipItems.length > 0) {
-                const item = tooltipItems[0];
-                ts = data.datasets[item.datasetIndex].data[item.index].x;
-              }
-              return ts;
-            }
-          },
           custom: (tooltip) => {
             chartTooltip.customTooltips(tooltip);
           }
         }
-      },
-      chartType: 'line'
-    };
+      } as ChartOptions
+    });
   }
 
-  ngOnChanges() {
-    if (!this.chart) {
-      return;
-    }
-
-    const lhsData = this.convert_timeseries(this.mdsCounter[this.lhsCounter]);
-    const rhsData = this.delta_timeseries(this.mdsCounter[this.rhsCounter]);
-
-    this.chart.datasets[0].data = lhsData;
-    this.chart.datasets[1].data = rhsData;
+  private updateChart() {
+    _.merge(this.chart, {
+      datasets: [
+        {
+          data: this.convert_timeseries(this.mdsCounter[this.lhsCounter])
+        },
+        {
+          data: this.delta_timeseries(this.mdsCounter[this.rhsCounter])
+        }
+      ] as ChartDataSets[]
+    });
   }
```

Comment:
Sorry to ask here. I created [an issue](https://tracker.ceph.com/issues/40096) about the chart is not automatically refreshed as time advanced. I can make it work if I add the following code:

```typescript
    this.chart.datasets = [...this.chart.datasets];
```

Is this related to change detection of Angular? Thanks!

Reply:
Yes, some times, especially when using external components, you have to force a change detection for Angular to refresh the data.

Reply:
@bk201 Thanks for the hint I will fix that :) Good spotting

Reply:
@bk201 done :)

Reply:
Thanks for fixing this too!